src/librustc_trans/trans/callee.rs

```diff
@@ -934,12 +934,13 @@ fn trans_args_under_call_abi<'blk, 'tcx>(
                     |srcval| {
                         adt::trans_field_ptr(bcx, repr_ptr, srcval, 0, i)
                     });
-                let arg_datum = arg_datum.to_expr_datum();
-                let arg_datum =
-                    unpack_datum!(bcx, arg_datum.to_rvalue_datum(bcx, "arg"));
-                let arg_datum =
-                    unpack_datum!(bcx, arg_datum.to_appropriate_datum(bcx));
-                llargs.push(arg_datum.add_clean(bcx.fcx, arg_cleanup_scope));
+                llargs.push(unpack_result!(bcx, {
+                    trans_arg_datum(bcx,
+                                    field_types[i],
```

Comment:
Couldn't this whole `for` loop be written as `llargs.extend(field_types.map(|field_ty| ...))`?